mllib/src/main/scala/org/apache/spark/ml/classification/Classifier.scala

```diff
@@ -62,6 +65,57 @@ abstract class Classifier[
   def setRawPredictionCol(value: String): E = set(rawPredictionCol, value).asInstanceOf[E]
 
   // TODO: defaultEvaluator (follow-up PR)
+
+  /**
+   * Extract [[labelCol]] and [[featuresCol]] from the given dataset,
+   * and put it in an RDD with strong types.
+   * @throws SparkException  if any label is not an integer >= 0
+   */
+  override protected def extractLabeledPoints(dataset: Dataset[_]): RDD[LabeledPoint] = {
+    dataset.select(col($(labelCol)).cast(DoubleType), col($(featuresCol))).rdd.map {
+      case Row(label: Double, features: Vector) =>
+        require(label % 1 == 0 && label >= 0, s"Classifier was given dataset with invalid label" +
```

Comment:
can you add a comment in the code for future developers that the number of classes has already been checked before with a call to `getNumClasses`?